PWGDQ/Core/VarManager.h

```diff
@@ -1027,6 +1051,106 @@ void VarManager::FillPairVertexing(C const& collision, T const& t1, T const& t2,
   }
 }
 
+template <uint32_t collFillMap, uint32_t fillMap, typename C, typename T1, typename T2, typename T3>
+void VarManager::FillDileptonLepton(C const& collision, T1 const& lepton1, T2 const& lepton2, T3 const& lepton3, float* values)
+{
+
+  constexpr bool eventHasVtxCov = ((collFillMap & Collision) > 0 || (collFillMap & ReducedEventVtxCov) > 0);
+  constexpr bool trackHasCov = ((fillMap & TrackCov) > 0 || (fillMap & ReducedTrackBarrelCov) > 0);
+  if (!values) {
+    values = fgValues;
+  }
+
+  float m1 = fgkMuonMass;
+  float m2 = fgkMuonMass;
+  float m3 = fgkMuonMass;
+  float mlepton = fgkMuonMass;
```

Comment:
this part needs to be generalized, on the lines of the suggestion i made for the general scope of this function